### ISSUE TITLE
[bitnami/redmine] The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable.

### ISSUE THREAD

**Issue by ytianxia6** [bitnami/charts#8084]
**Which chart**:
The name (and version) of the affected chart
bitnami/redmine  17.0.7

**Describe the bug**
the mariadb pod cannot start!

**To Reproduce**

installed the helm with values:

``` yaml
affinity: {}
allowEmptyPassword: false
args: []
autoscaling:
  enabled: false
  maxReplicas: 11
  minReplicas: 1
  targetCPU: 50
  targetMemory: 50
certificates:
  customCA: []
  customCertificate:
    certificateLocation: /etc/ssl/certs/ssl-cert-snakeoil.pem
    certificateSecret: ''
    chainLocation: /etc/ssl/certs/mychain.pem
    chainSecret:
      key: ''
      name: ''
    keyLocation: /etc/ssl/private/ssl-cert-snakeoil.key
  extraEnvVars: []
  image:
    pullPolicy: IfNotPresent
    pullSecrets: []
    registry: docker.io
    repository: bitnami/bitnami-shell
    tag: 10-debian-10-r203
command: []
commonAnnotations: {}
commonLabels: {}
containerPort: 3000
containerSecurityContext:
  enabled: false
  runAsUser: 1001
customLivenessProbe: {}
customPostInitScripts: {}
customReadinessProbe: {}
customStartupProbe: {}
databaseType: mariadb
existingSecret: ''
externalDatabase:
  database: redmine
  existingSecret: ''
  host: redmine-db-mysql.redmine.svc.cluster.local
  password: *****
  port: 3306
  user: root
extraDeploy: []
extraEnvVars: []
extraEnvVarsCM: ''
extraEnvVarsSecret: ''
extraVolumeMounts: []
extraVolumes: []
fullnameOverride: ''
global:
  imagePullSecrets: []
  imageRegistry: ''
  storageClass: ''
hostAliases: []
image:
  debug: false
  pullPolicy: IfNotPresent
  pullSecrets: []
  registry: docker.io
  repository: bitnami/redmine
  tag: 4.2.2-debian-10-r13
ingress:
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt-prod
    kubernetes.io/ingress.class: nginx
  apiVersion: ''
  certManager: true
  enabled: true
  extraHosts: []
  extraPaths: []
  extraTls: []
  hostname: xxx
  path: /
  pathType: ImplementationSpecific
  secrets: []
  selfSigned: false
  tls: true
initContainers: []
kubeVersion: ''
livenessProbe:
  enabled: true
  failureThreshold: 6
  initialDelaySeconds: 300
  path: /
  periodSeconds: 10
  successThreshold: 1
  timeoutSeconds: 5
mailReceiver:
  affinity: {}
  allowOverride: ''
  args:
    - '-c'
    - /cj/receive.sh
  assignedTo: ''
  category: ''
  command:
    - bash
  containerSecurityContext:
    enabled: false
    runAsNonRoot: true
    runAsUser: 1001
  defaultGroup: ''
  enabled: false
  extraEnvVars: []
  extraEnvVarsCM: ''
  extraEnvVarsSecret: ''
  extraVolumeMounts: []
  extraVolumes: []
  host: ''
  imapFolder: INBOX
  initContainers: []
  mailProtocol: IMAP
  moveOnFailure: ''
  moveOnSuccess: ''
  noAccountNotice: 1
  noPermissionCheck: 0
  nodeAffinityPreset:
    key: ''
    type: ''
    values: []
  nodeSelector: {}
  password: ''
  podAffinityPreset: ''
  podAnnotations: {}
  podAntiAffinityPreset: soft
  podLabels: {}
  podSecurityContext:
    enabled: true
    fsGroup: 1001
  port: 993
  priority: ''
  priorityClassName: ''
  project: ''
  projectFromSubaddress: ''
  schedule: '*/5 * * * *'
  sidecars: []
  ssl: true
  startTLS: false
  status: ''
  suspend: true
  tolerations: []
  tracker: ''
  unknownUserAction: ignore
  username: ''
mariadb:
  architecture: standalone
  auth:
    database: redmine
    existingSecret: ''
    password: ''
    rootPassword: ''
    username: bn_redmine
  enabled: true
  primary:
    persistence:
      accessModes:
        - ReadWriteOnce
      enabled: true
      existingClaim: ''
      hostPath: ''
      size: 8Gi
      storageClass: ''
nameOverride: ''
nodeAffinityPreset:
  key: ''
  type: ''
  values: []
nodeSelector: {}
persistence:
  accessModes:
    - ReadWriteOnce
  dataSource: {}
  enabled: true
  existingClaim: ''
  size: 80Gi
  storageClass: ''
podAffinityPreset: ''
podAnnotations: {}
podAntiAffinityPreset: soft
podDisruptionBudget:
  create: false
  maxUnavailable: ''
  minAvailable: ''
podLabels: {}
podSecurityContext:
  enabled: false
  fsGroup: 1001
postgresql:
  enabled: false
  existingSecret: ''
  persistence:
    accessMode: ReadWriteOnce
    enabled: true
    existingClaim: ''
    size: 8Gi
    storageClass: ''
  postgresqlDatabase: bitnami_redmine
  postgresqlPassword: ''
  postgresqlUsername: bn_redmine
readinessProbe:
  enabled: true
  failureThreshold: 6
  initialDelaySeconds: 5
  path: /
  periodSeconds: 10
  successThreshold: 1
  timeoutSeconds: 5
redmineEmail: xxx
redmineLanguage: cn
redminePassword: ********
redmineUsername: admin
replicaCount: 1
resources:
  limits: {}
  requests:
    cpu: 300m
    memory: 512Mi
schedulerName: ''
service:
  annotations: {}
  clusterIP: ''
  externalTrafficPolicy: Cluster
  extraPorts: []
  loadBalancerIP: ''
  loadBalancerSourceRanges: []
  nodePort: ''
  port: 80
  type: ClusterIP
serviceAccount:
  create: false
  name: ''
sidecars: []
smtpExistingSecret: ''
smtpHost: ''
smtpPassword: ''
smtpPort: ''
smtpProtocol: ''
smtpUser: ''
startupProbe:
  enabled: false
  failureThreshold: 6
  initialDelaySeconds: 300
  path: /
  periodSeconds: 10
  successThreshold: 1
  timeoutSeconds: 5
tolerations: []
updateStrategy:
  rollingUpdate: {}
  type: RollingUpdate
volumePermissions:
  enabled: false
  resources:
    limits: {}
    requests: {}
  securityContext:
    runAsUser: 0
```

the pod mariadb logs:

``` bash
mariadb 05:39:43.07 
mariadb 05:39:43.07 Welcome to the Bitnami mariadb container
mariadb 05:39:43.07 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 05:39:43.07 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 05:39:43.08 
mariadb 05:39:43.08 INFO  ==> ** Starting MariaDB setup **
mariadb 05:39:43.12 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 05:39:43.14 INFO  ==> Initializing mariadb database
mariadb 05:39:43.16 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mariadb 05:39:43.16 INFO  ==> Using persisted data
mariadb 05:39:43.21 INFO  ==> Running mysql_upgrade
mariadb 05:39:43.22 INFO  ==> Starting mariadb in background
```

**Expected behavior**
mariadb start normal.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.3", GitCommit:"d506314abfb5d21419df8c7e7e68012379db2354", GitTreeState:"clean", GoVersion:"go1.16.5"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:45:37Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:39:34Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"linux/amd64"}```

**Additional context**
Add any other context about the problem here.

**Comment by juan131** [comment]
Hi @ytianxia6 

The message below is just a warning and it's an expected one in the MariaDB chart:

```
WARN  ==>  The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
```

We need to figure out why MariaDB doesn't start and the logs get stuck at:

```
INFO  ==> Starting mariadb in background
```

Could you please install the chart using the flag `--set image.debug=true`? That will increase the logs verbosity and we'll have more info about what's going on.
Could you also share the output of running `kubectl describe pod MARIADB_POD_NAME` (where MARIADB_POD_NAME is a placeholder to be substituted with the actual name of the MariaDB pod)?

Finally, please take a look to our Troubleshooting guide and the solutions we provide there to some common issues using Bitnami charts:

- https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/

**Comment by ytianxia6** [comment]
> Hi @ytianxia6
> 
> The message below is just a warning and it's an expected one in the MariaDB chart:
> 
> ```
> WARN  ==>  The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
> ```
> 
> We need to figure out why MariaDB doesn't start and the logs get stuck at:
> 
> ```
> INFO  ==> Starting mariadb in background
> ```
> 
> Could you please install the chart using the flag `--set image.debug=true`? That will increase the logs verbosity and we'll have more info about what's going on. Could you also share the output of running `kubectl describe pod MARIADB_POD_NAME` (where MARIADB_POD_NAME is a placeholder to be substituted with the actual name of the MariaDB pod)?
> 
> Finally, please take a look to our Troubleshooting guide and the solutions we provide there to some common issues using Bitnami charts:
> 
> * https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/

kubectl describe pod returns:

```
Name:         redmine-mariadb-0
Namespace:    redmine
Priority:     0
Node:         k8s.node2/192.168.79.23
Start Time:   Thu, 11 Nov 2021 08:35:48 +0000
Labels:       app.kubernetes.io/component=primary
              app.kubernetes.io/instance=redmine
              app.kubernetes.io/managed-by=Helm
              app.kubernetes.io/name=mariadb
              controller-revision-hash=redmine-mariadb-7688d4bcb8
              helm.sh/chart=mariadb-9.4.4
              statefulset.kubernetes.io/pod-name=redmine-mariadb-0
Annotations:  checksum/configuration: e5224c195e9b3ad899d67e931d34cbcd15b72418f7c1994b67e2953203a4e891
              cni.projectcalico.org/containerID: 75cac1939be40202091944530e8dc0daf42007b096cd3cbdf57a71b072bcf4d1
              cni.projectcalico.org/podIP: 10.48.37.229/32
              cni.projectcalico.org/podIPs: 10.48.37.229/32
Status:       Running
IP:           10.48.37.229
IPs:
  IP:           10.48.37.229
Controlled By:  StatefulSet/redmine-mariadb
Containers:
  mariadb:
    Container ID:   docker://f19b9c95b1821c8a310e7f81eb54e6ddf8357b159feb5ae8725eea9205b0ce10
    Image:          docker.io/bitnami/mariadb:10.5.12-debian-10-r19
    Image ID:       docker-pullable://bitnami/mariadb@sha256:8084bda74e4ce3f2477e098639fd6299a386cf430bbea2a0a39794c1a33a9c69
    Port:           3306/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Thu, 11 Nov 2021 08:40:30 +0000
    Last State:     Terminated
      Reason:       Error
      Exit Code:    137
      Started:      Thu, 11 Nov 2021 08:37:35 +0000
      Finished:     Thu, 11 Nov 2021 08:40:28 +0000
    Ready:          False
    Restart Count:  1
    Liveness:       exec [/bin/bash -ec password_aux="${MARIADB_ROOT_PASSWORD:-}"
if [[ -f "${MARIADB_ROOT_PASSWORD_FILE:-}" ]]; then
    password_aux=$(cat "$MARIADB_ROOT_PASSWORD_FILE")
fi
mysqladmin status -uroot -p"${password_aux}"
] delay=120s timeout=1s period=10s #success=1 #failure=3
    Readiness:  exec [/bin/bash -ec password_aux="${MARIADB_ROOT_PASSWORD:-}"
if [[ -f "${MARIADB_ROOT_PASSWORD_FILE:-}" ]]; then
    password_aux=$(cat "$MARIADB_ROOT_PASSWORD_FILE")
fi
mysqladmin status -uroot -p"${password_aux}"
] delay=30s timeout=1s period=10s #success=1 #failure=3
    Environment:
      BITNAMI_DEBUG:          false
      MARIADB_ROOT_PASSWORD:  <set to the key 'mariadb-root-password' in secret 'redmine-mariadb'>  Optional: false
      MARIADB_USER:           bn_redmine
      MARIADB_PASSWORD:       <set to the key 'mariadb-password' in secret 'redmine-mariadb'>  Optional: false
      MARIADB_DATABASE:       redmine
    Mounts:
      /bitnami/mariadb from data (rw)
      /opt/bitnami/mariadb/conf/my.cnf from config (rw,path="my.cnf")
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-xwbz4 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  data-redmine-mariadb-0
    ReadOnly:   false
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      redmine-mariadb
    Optional:  false
  kube-api-access-xwbz4:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                 From               Message
  ----     ------     ----                ----               -------
  Normal   Scheduled  5m51s               default-scheduler  Successfully assigned redmine/redmine-mariadb-0 to k8s.node2
  Normal   Pulled     4m4s                kubelet            Container image "docker.io/bitnami/mariadb:10.5.12-debian-10-r19" already present on machine
  Normal   Created    4m1s                kubelet            Created container mariadb
  Normal   Started    4m                  kubelet            Started container mariadb
  Warning  Unhealthy  97s (x3 over 117s)  kubelet            Liveness probe failed: mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/opt/bitnami/mariadb/tmp/mysql.sock' (111)'
Check that mysqld is running and that the socket: '/opt/bitnami/mariadb/tmp/mysql.sock' exists!
  Normal   Killing    97s                   kubelet  Container mariadb failed liveness probe, will be restarted
  Warning  Unhealthy  67s (x18 over 3m27s)  kubelet  Readiness probe failed: mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/opt/bitnami/mariadb/tmp/mysql.sock' (111)'
Check that mysqld is running and that the socket: '/opt/bitnami/mariadb/tmp/mysql.sock' exists!
```

I use the bitnami/redmine helm package, I cannot find where to set the mariadb image debug to true, bug only the redmine image. if I set the redmine image's debug to true, the log is same .

**Comment by juan131** [comment]
Hi @ytianxia6 

The error reported in the readiness/liveness probes suggest me that MariaDB couldn't be started properly (for whatever reason) but I don't have enough info about the reason:

```console
mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local MySQL server through socket '/opt/bitnami/mariadb/tmp/mysql.sock' (111)'
Check that mysqld is running and that the socket: '/opt/bitnami/mariadb/tmp/mysql.sock' exists!
```

Let's see if we can obtain more information once you're able to enable debug logs in MariaDB.

> I use the bitnami/redmine helm package, I cannot find where to set the mariadb image debug to true, bug only the redmine image. if I set the redmine image's debug to true, the log is same .

MariaDB is a subchar of Redmine, therefore you can also modify its parameters. Please try to upgrade your release appending the content below to your custom **values.yaml** (you can also use the flag `--set mariadb.image.debug=true`):

```yaml
mariadb:
  image:
    debug: true
```

**Comment by ytianxia6** [comment]
after I set the debug to true, the logs:

```
mariadb 07:24:22.35 
mariadb 07:24:22.35 Welcome to the Bitnami mariadb container
mariadb 07:24:22.35 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 07:24:22.36 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 07:24:22.36 
mariadb 07:24:22.36 INFO  ==> ** Starting MariaDB setup **
mariadb 07:24:22.41 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 07:24:22.44 INFO  ==> Initializing mariadb database
mariadb 07:24:22.44 DEBUG ==> Ensuring expected directories/files exist
mariadb 07:24:22.47 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mariadb 07:24:22.47 INFO  ==> Using persisted data
mariadb 07:24:22.53 INFO  ==> Running mysql_upgrade
mariadb 07:24:22.54 INFO  ==> Starting mariadb in background
2021-11-12  7:24:22 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.5.12-MariaDB) starting as process 48 ...
2021-11-12  7:24:22 0 [Note] InnoDB: Uses event mutexes
2021-11-12  7:24:22 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-11-12  7:24:22 0 [Note] InnoDB: Number of pools: 1
2021-11-12  7:24:22 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2021-11-12  7:24:22 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
2021-11-12  7:24:22 0 [Note] InnoDB: Using Linux native AIO
2021-11-12  7:24:22 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
2021-11-12  7:24:22 0 [Note] InnoDB: Completed initialization of buffer pool
2021-11-12  7:24:22 0 [Note] InnoDB: 128 rollback segments are active.
2021-11-12  7:24:22 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2021-11-12  7:24:22 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2021-11-12  7:24:22 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2021-11-12  7:24:23 0 [Note] InnoDB: 10.5.12 started; log sequence number 45090746; transaction id 14461
2021-11-12  7:24:23 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-11-12  7:24:23 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
2021-11-12  7:24:23 0 [Note] InnoDB: Buffer pool(s) load completed at 211112  7:24:23
2021-11-12  7:24:23 0 [Note] Server socket created on IP: '127.0.0.1'.
2021-11-12  7:24:23 0 [ERROR] Fatal error: Can't open and lock privilege tables: Table 'mysql.db' doesn't exist
2021-11-12  7:24:23 0 [ERROR] Aborting
```

**Comment by juan131** [comment]
Hi @ytianxia6 

Although I'm not 100% sure, however it looks like following happened when you installed the chart:

- During the 1st boot of the MariaDB container, it took longer than expected to initialize MariaDB and the readiness/liveness probes failed too many times **restarting MariaDB before the initialisation was completed.**
- After that, MariaDB reuses the existing data (see the `INFO  ==> Using persisted data` message), the data that was created during the 1st boot, but this data seems to be in an inconsistent status. Therefore it's unable to start and you see the `Table 'mysql.db' doesn't exist` error.

So, let's try the following thing to solve the issue using this workaround:

> Note: ONLY follow this workaround if you hadn't started to add any data to your Redmine setup.

- First, uninstall the Redmine release and delete the MariaDB PVC:

```console
$ helm uninstall redmine
$ kubectl delete pvc data-redmine-mariadb-0
```

- Then, install Redmine again, but this time let's give much more time for MariaDB to initialize (just in case your K8s environment require it):

```console
$ helm install redmine bitnami/redmine --set mariadb.image.debug=true --set mariadb.primary.readinessProbe.initialDelaySeconds=90
```

- Finally, let's wait for Redmine and MariaDB (monitoring their logs) and see if they're able to reach the "ready" status

**Comment by ytianxia6** [comment]
But I have migrated the data to the new redmine, and it has been running normally for some time.
This is a problem after restarting.

**Comment by juan131** [comment]
All right, then don't follow the workaround, I don't want you to lose any data.

Question: did you perform any "helm upgrade" recently?

**Comment by ytianxia6** [comment]
Yes, I did. but I did it only after mariadb cannot startup, and  try to find out the error.

**Comment by juan131** [comment]
Hi @ytianxia6 

Please correct me if I'm wrong but this is the timeline of events:

- You installed a release of the Bitnami Redmine chart (`helm install ... bitnami/redmine`).
- Then, you imported your data (from a previous installation you had) in your new Redmine release.
- It was working fine for a while.
- Then you restarted/recreated your containers (for whatever reason, e.g. rescheduling pods).
- MariaDB started to fail (unable to iniatialize).
- Then, you upgraded your release doing a `helm upgrade ...`.

It's very likely that the problem is simply related to the probes because you imported a very bit database so MariaDB is taking too much time during the initialization and the probes fail and restart the container before it's fully initialized.

You could try to do a new "helm upgrade" increasing the "initialDelaySeconds" for MariaDB probes but please ensure you set the `mariadb.auth.rootPassword` and `mariadb.auth.password` parameters with the original password values during the upgrade.

**Comment by ytianxia6** [comment]
Yes, the timeline is all right.
I tried to helm with the `mariadb.auth.rootPassword` and `mariadb.auth.password` in the secret `redmine-mariadb` ，the error goes as before.
Is it modify the secret while `helm upgrade`? I didn't save the password before!
so Is there any way to solve it?

**Comment by juan131** [comment]
> I tried to helm with the mariadb.auth.rootPassword and mariadb.auth.password in the secret redmine-mariadb, the error goes as before.

Let's go a step further and perform the upgrade but this time let's disable the probes for MariaDB (so it has all the time it needs to initialize):

```console
$ export MARIADB_ROOT_PASSWORD=$(kubectl get secret --namespace default redmine-mariadb -o jsonpath="{.data.mariadb-root-password}" | base64 --decode)
$ export MARIADB_PASSWORD=$(kubectl get secret --namespace default redmine-mariadb -o jsonpath="{.data.mariadb-password}" | base64 --decode)
$ export REDMINE_PASSWORD=$(kubectl get secret --namespace default redmine -o jsonpath="{.data.redmine-password}" | base64 --decode)
$ helm upgrade redmine bitnami/redmine \
  --set redminePassword=$REDMINE_PASSWORD \
  --set mariadb.image.debug=true \
  --set mariadb.auth.rootPassword=$MARIADB_ROOT_PASSWORD \
  --set mariadb.auth.password=$MARIADB_PASSWORD \
  --set mariadb.primary.readinessProbe.enabled=false \
  --set mariadb.primary.livenessProbe.enabled=false
```

After that, monitor MariaDB logs and wait for MariaDB to initialize.

> Is it modify the secret while helm upgrade? I didn't save the password before!
so Is there any way to solve it?

Yes, you can always [perform a helm rollback as it's explained in the Troubleshooting Guide](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#solve-issue).

**Comment by ytianxia6** [comment]
as the step, the logs stops at
``` 
ariadb 09:41:26.97 Welcome to the Bitnami mariadb container
mariadb 09:41:26.98 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb 09:41:26.98 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb 09:41:26.98 
mariadb 09:41:26.98 INFO  ==> ** Starting MariaDB setup **
mariadb 09:41:27.02 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 09:41:27.04 INFO  ==> Initializing mariadb database
mariadb 09:41:27.06 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file.
mariadb 09:41:27.06 INFO  ==> Using persisted data
mariadb 09:41:27.09 INFO  ==> Running mysql_upgrade
mariadb 09:41:27.10 INFO  ==> Starting mariadb in background

```

just as top!
so what happend?

**Comment by ytianxia6** [comment]
@juan131 Thank you very much for your help! I discarded the data and did a re-migrated to solve the problem!
this issue can be closed!

**Comment by juan131** [comment]
Great!!! I'm very glad you were able to solve the issue by giving more time to MariaDB to initialize the database!

**Comment by tanzwut2525** [comment]
I got the same problem. 
I`ve changed  line: "https://github.com/bitnami/charts/blob/da7bc20d343cdef705f7ebc50ef8b0046e6eb0e0/bitnami/mysql/values.yaml#L181"
From: "collation-server=utf8_general_ci" to  "    collation-server=utf8mb4_general_ci"
As a result I got this msg: 
"The mysql configuration file '/opt/bitnami/mysql/conf/my.cnf' is not writable. Configurations based on environment variables will not be applied for this file." and then POD with mysql was crashed and restarted a few times.

So, looks like by some reason I cant change encoding.
